### ISSUE TITLE
runCustomValidations() and show errors

### ISSUE THREAD

**Issue by lhwong** [sir-dunxalot/ember-easy-form-extensions#44]
Hi,

My server returns the attributes of all of the fields. Based on those attributes (e.g. field type, label), the fields are rendered (see the template at the bottom of the page). This works pretty well.
However, I face issue on validation. I don't know how to specify the field in the validations like the following.

```
validations: {
    'model.name': {
      presence: true
    }
  },
```

Hence, my only option seems to be runCustomValidations(). If using this, how can I show the errors next to the field?

```
runCustomValidations() {
    return new Ember.RSVP.Promise(function(resolve, reject) {
      //Run some kind of validations Ember.Validations doesn't support
      //resolve() to validate the model successfully
      //reject() to validate the model unsuccessfully

      var fieldRequired = false;

      this.model.get('fields').forEach(function(field) {
          if (field.get('fieldRequired')) {
            if (Ember.isEmpty(field.get('value'))) {
                fieldRequired = true;
            } 
          }
        });  

        if (fieldRequired) {
            reject(); 
        } else {
            resolve();
        }   
    });
  },
```

```
{{#form-wrapper}}

  {{#form-controls}}
    {{#each field in model.fields}}
       {{#if field.hasContent}} 
            {{input-group value=field.value prompt="Please select one" label=field.label type="select" inputClass="form-control" content=field.content}}
        {{else}}
          {{!#if field.isMultipleSubform}}
            {{!field.label}}
            {{!jqui-autocomplete value=field.value source="/api/1/hubs" inputClass="form-control"}} 
          {{!else}}
            {{#if field.isHTMLLabel}}
              {{{field.label}}}
            {{else if field.isTelephoneNo}}
              {{input-group value=field.value  label=field.label type='tel' inputClass="form-control"}}
            {{else if field.isMobileTelephoneNo}}
              {{input-group value=field.value  label=field.label type='telmobile' inputClass="form-control"}}
            {{else if field.isState}}
              {{input-group value=field.value prompt="Please select one" label=field.label type="select" inputClass="form-control" content=states}} 
            {{else if field.isCountry}}
              {{input-group value=field.value prompt="Please select one" label=field.label type="select" inputClass="form-control" content=countries}}
            {{else if field.isLanguage}}
              {{input-group value=field.value prompt="Please select one" label=field.label type="select" inputClass="form-control" content=languages}} 
            {{else if field.isRace}}
              {{input-group value=field.value prompt="Please select one" label=field.label type="select" inputClass="form-control" content=races}}         
            {{else if field.isSubform}}
              {{#each rows in field.multipleSubform}}
              <div class="row">
                {{!#each row in rows}}
                  <div class="col-md-6">
                      {{input-group value=rows.[0].value label=rows.[0].label type=rows.[0].inputType inputClass="form-control"}} 

                  </div>
                {{!/each}}
                <div class="col-md-6">

                     {{input-group value=rows.[1].value label=rows.[1].label prompt="Please select one" type="select" inputClass="form-control" content=rows.[1].content}} 

                  </div>
              </div>  
              {{/each}}
              <p>
              <button {{action "addElement" field}}>Add identifier</button>
              </p>
            {{else if field.isInputDate}}
              {{input-group value=field.value  label=field.label type="datepicker" inputClass="form-control"}}     
            {{else}}

              {{input-group value=field.value  label=field.label type=field.inputType inputClass="form-control"}} 
            {{/if}}  
          {{!/if}}
        {{/if}}
    {{/each}}    
  {{/form-controls}}


  {{#if model.isReady}}
    {{form-submission

      cancel=true 
      cancelText='Cancel'
      submit=true 
      submitText='Claim' 
    }}
  {{/if}}

{{/form-wrapper}}
```

Regards,
--Nick

**Comment by sir-dunxalot** [comment]
Hi Nick,

The validations are handled by the [Ember Validations library](https://github.com/dockyard/ember-validations) so you'll need to reference that documentation. To show an error for a particular object you push the error to `errors.propertyName`. For example, `this.get('errors.name').pushObject('should not be blank');`.

However, you could store the object anywhere you like (e.g. on each `model.get('fields')` item)  and use the `{{error-field}}` inside the above `{{each}}` helper to render the errors.

**Comment by sir-dunxalot** [comment]
This is possible by adding logic to the `onInvalidSubmission()` method, added with #48 and released in `2.0.0`.
